eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,160 @@
+parameters:
+  - name: SpecRepoBranch
+    type: string
+    default: 'main'
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoOwner
+    type: string
+  - name: SdkRepoName
+    type: string
+
+  - name: SdkRepoNameForSpecGen
```

Comment:
This parameter can be computed by the `SdkRepoUrl`. for example, `sdkRepoNameForSpecGen` will be 'azure-sdk-for-java' if the `SdkRepoUrl` is 'https://github.com/azure/azure-sdk-for-java'. The only exception is 'azure-sdk-for-net' repo, the value will be 'azure-sdk-for-net-track2'. We will make 'azure-sdk-for-net' consistent with other repos when roll out this pipeline to azure rest api spec repo.

Reply:
I added the extra parameter specifically for the dotnet case.

Reply:
In my opinion passing the parameter parts and combining them later is easier and neater than adding an extra step to generate the parameters from the full url. Also, it's easier to just pass an extra parameter to handle the specific dotnet case rather than adding a condition for dotnet in the shared pipeline definition.

Reply:
I generally agree passing the parts is easier but we should also think about the usages here. Do we expect the user to ever set these or will there be a wrapper pipeline that hardcodes them?

Reply:
They will be set replaced) programmatically from a configuration in the PR trigger case. So we agreed it best to use the url and split it later on in the logic